pkg/encoding/encoding.go

```diff
@@ -18,167 +20,606 @@ import (
-	"iso88591":  charmap.ISO8859_1,
-	"ibm037":    charmap.CodePage037,
-	"ibm437":    charmap.CodePage437,
-	"ibm850":    charmap.CodePage850,
-	"ibm852":    charmap.CodePage852,
-	"ibm855":    charmap.CodePage855,
-	"ibm858":    charmap.CodePage858,
-	"ibm860":    charmap.CodePage860,
-	"ibm862":    charmap.CodePage862,
-	"ibm863":    charmap.CodePage863,
-	"ibm865":    charmap.CodePage865,
-	"ibm1047":   charmap.CodePage1047,
-	"ibm1140":   charmap.CodePage1140,
-	"iso88596e": charmap.ISO8859_6E,
-	"iso88596i": charmap.ISO8859_6I,
-	"iso88598e": charmap.ISO8859_8E,
-	"iso88598i": charmap.ISO8859_8I,
-	"iso88599":  charmap.ISO8859_9,
-	"hzgb2312":  simplifiedchinese.HZGB2312,
-	// Not https://github.com/golang/text/blob/HEAD/encoding/ianaindex/ianaindex.go#L156 :
-	"macintoshcyrillic": charmap.MacintoshCyrillic,
-	// Not in https://github.com/golang/text/blob/HEAD/encoding/htmlindex/map.go#L64 or
-	//        https://github.com/golang/text/blob/HEAD/encoding/ianaindex/ianaindex.go#L156 :
-	"utf8bom": unicode.UTF8,
-	"utf32be": utf32.UTF32(utf32.BigEndian, utf32.IgnoreBOM),
-	"utf32le": utf32.UTF32(utf32.LittleEndian, utf32.IgnoreBOM),
-}
-
-// In https://github.com/golang/text/blob/HEAD/encoding/ianaindex/ianaindex.go#L156
-// but not included above:
-// 	 enc3:    asciiEnc,
-//   enc1015: unicode.UTF16(unicode.BigEndian, unicode.UseBOM)
-
-// DecodeBytes converts a byte array to a string
-func DecodeBytes(contentBytes []byte) (string, string, error) {
-	contentString := string(contentBytes)
+const (
+	// HitMissRatioForUTF1632Checks defines the ratio of hits to misses that must
+	// be met for the IsUTF* functions to return true. For example, the number 1.1
+	// requires the function find 10% more hits than misses. This number was chosen
+	// empiricallly, but may need to be adjusted as we gather more test results.
+	HitMissRatioForUTF1632Checks = 1.01
 
-	charset, err := detectText(contentBytes)
-	if err != nil {
-		if IsBinaryFile(contentBytes) {
-			return contentString, BinaryData, nil
+	// HitMissRatioIfBOMFound is returned if a file begins with a BOM
+	// (Byte Order Mark).
+	HitMissRatioIfBOMFound = 1023.0
+
+	// MinConfidenceForUTF1632Checks defines the minimum confidence factor to use
+	// our own check for if a file is UTF16/UTF32 encoded. 100 represents
+	// a 100% confidence level, which is the highest confidence returned by our
+	// chardet library.
+	// testdata/wpt/resources/utf-32-big-endian-nobom.html is an 85.0.
+	MinConfidenceForUTF1632Checks = 90.0
+
+	// BinaryData contains the string to return if the data contains binary
+	// data (data that is not decodable, by any of the decoders that golang provides).
+	BinaryData = "binary"
+
+	// UnknownEncoding is returned if the encoding could not be determined.
+	UnknownEncoding = "unknown"
+
+	// See https://spec.editorconfig.org/#supported-pairs
+	// CharsetUnset defines the value allowing for file encoding.
```

Comment:
We could use the constants at https://github.com/editorconfig/editorconfig-core-go/blob/7404f9a31780afcfa5bc3a62697476e720b39c5d/editorconfig.go#L36 instead.